llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -893,18 +896,27 @@ static bool isCmpSameOrSwapped(const CmpInst *BaseCI, const CmpInst *CI,
 static InstructionsState getSameOpcode(ArrayRef<Value *> VL,
                                        const TargetLibraryInfo &TLI) {
   // Make sure these are all Instructions.
-  if (!all_of(VL, IsaPred<Instruction>))
+  if (!all_of(VL, IsaPred<Instruction, PoisonValue>))
+    return InstructionsState::invalid();
+
+  auto *It = find_if(VL, IsaPred<Instruction>);
+  if (It == VL.end())
+    return InstructionsState::invalid();
```

Comment:
can this realistically happen given the all_of check above? replace with assert?

Reply:
It can happen. The check above just checks that all elements are Instructions or PoisonValues, but does not check that there is at least single Instruction in the ArrayRef